src/main.ts

```diff
@@ -147,18 +147,41 @@ export class Purchases {
   }
 
   /**
-   * Configures the Purchases SDK. This should be called as soon as your app
-   * has a unique user id for your user. You should only call this once, and
+   * Configures the Purchases SDK with app-provided user identifiers. This should be called as
+   * soon as your app has a unique user id for your user. You should only call this once, and
    * keep the returned instance around for use throughout your application.
    * @param apiKey - RevenueCat API Key. Can be obtained from the RevenueCat dashboard.
    * @param appUserId - Your unique id for identifying the user.
-   * @param httpConfig - Advanced http configuration to customise the SDK usage {@link HttpConfig}.
+   * @param httpConfig - Optional. Advanced http configuration to customise the SDK usage.
    * @throws {@link PurchasesError} if the API key or user id are invalid.
+   * @remarks For RevenueCat-managed identifiers, use configure with AppUserIDProvider.RevenueCat
    */
   static configure(
     apiKey: string,
-    appUserId: string | IdentityMode,
-    httpConfig: HttpConfig = defaultHttpConfig,
+    appUserId: string,
```

Comment:
I think having 2 overrides for this method is overkill.

In the end we need to handle 2 cases:
1. The appUserId is passed
2. The appUserId is not passed.

Proposal
```

class Purchases{

  static final GeneratedByRevenueCat='Purchases.GeneratedByRevenueCat'

  static configure(
     apiKey:string,
     appUserId: string ,
     httpConfig?: HttpConfig
  ) {
     
     
     if(appUserId === Purchases.GeneratedByRevenueCat){
        // generate the app user id
     } else {
       // use whatever was passed.
     }
     
  }
```

Then we can just use it:
```
// anonymous
Purchases.configure(API_KEY, Purchases.GeneratedByRevenueCat)

// non anonymous
Purchases.configure(API_KEY, 'my_custom_app_user_id')
```

Reply:
Sorry for being so late to this.
The proposal makes sense at a high level, although I'd suggest making the name more explicit. 

`Purchases.GeneratedByRevenueCat` sounds like you're saying that "the purchases are generated by RevenueCat" instead of "the app user ids are generated by RevenueCat", so I'd try to make sure that the name appUserID is there explicitly, like `Purchases.AutoGeneratedAppUserID` or something.